tsdb/head.go

```diff
@@ -1360,10 +1362,13 @@ func (h *Head) indexRange(mint, maxt int64) *headIndexReader {
 
 // Chunks returns a ChunkReader against the block.
 func (h *Head) Chunks() (ChunkReader, error) {
-	return h.chunksRange(math.MinInt64, math.MaxInt64, h.iso.State()), nil
+	return h.chunksRange(math.MinInt64, math.MaxInt64, h.iso.State())
 }
 
-func (h *Head) chunksRange(mint, maxt int64, is *isolationState) *headChunkReader {
+func (h *Head) chunksRange(mint, maxt int64, is *isolationState) (*headChunkReader, error) {
+	if h.closed {
```

Comment:
How is this exactly helping here? Because IIUC, there can be a race between calling this and `Close()`, we could get past this check while it is being closed simultaneously.

Reply:
I see the issue description where it is helping :). Is there any possibility of race as I mentioned? Or is it expected to not be accessed when we are closing?

Reply:
I don't think there is a race but since this is not a hot path it is just safer to add a mutex so I updated the PR.